### PR TITLE
CVE-2014-3581

### PR DESCRIPTION
**Pull request by yerngpaii**
updated yml file answering questions

### CODE REVIEW

File: cves/CVE-2014-3581.yml [hunk 1]
```diff
@@ -66,10 +69,9 @@ fixes_vcc_instructions: |
...
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
-   - commit: c164ca7383d5f204915d85a5826655d3f1557148
-     note:
-   - commit:
-     note:
+   - commit: 2fa69ec09458a7a6519eb1e960b1fee45da66143
+     note: |
+      Fixed a mod_cache NULL pointer deference when handling empty Content-Type.
 vccs:
```
Comment (Ramenous) [inline cves/CVE-2014-3581.yml:75]: There seems to be no vccs for this vulnerability. Not sure if it was missed or intentionally left empty, but a note or some kind should be provided as this is pretty important.

**Review by Ramenous — CHANGES_REQUESTED** [review]
For the lessons section:
I think this vulnerability applies to two of these lessons:
1. Denial of service: since the description describes this vulnerability as a denial of service due to attackers being able to crash the application with empty content type headers
2. Distrust Input: since the application doesn't check if the content type headers are empty, that allows attackers to be able to supply empty content type headers and crash the application.

Overall, this is an interesting vulnerability because I've never heard of using empty headers to crash an application: +5 points!!

File: cves/CVE-2014-3581.yml (hunk 1 above)
Comment (yerngpaii) [reply cves/CVE-2014-3581.yml:75]: noted and added the vcc, it took a while to find. The commit was made in 2008 while the fix was implemented in 2014

**Comment by yerngpaii** [comment]
@Ramenous changes were made, can you please review this again for anything I may have missed?

**Comment by Ramenous** [comment]
> @Ramenous changes were made, can you please review this again for anything I may have missed?

I think you're good with this.

File: cves/CVE-2014-3581.yml [hunk 2]
```diff
@@ -48,7 +48,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  There was a flaw in how a module handled Content-Type headers. The service would crash
+  and lead to a denial of service attack if attackers would supplied an empty Content-Type
```
Comment (CraftSpider) [inline cves/CVE-2014-3581.yml]:
```suggestion
  and lead to a denial of service attack if attackers supplied an empty Content-Type 
```

File: cves/CVE-2014-3581.yml [hunk 3]
```diff
@@ -48,7 +48,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  There was a flaw in how a module handled Content-Type headers. The service would crash
+  and lead to a denial of service attack if attackers would supplied an empty Content-Type 
+  header, since it would be a NULL pointer dereference and leads to application crash.
```
Comment (CraftSpider) [inline cves/CVE-2014-3581.yml]:
```suggestion
  header, since it would be a NULL pointer dereference and lead to an application crash.
```

**Review by CraftSpider — COMMENTED** [review]
I generally don't have comments, but I marked two sentences for readability

**Review by Ramenous — APPROVED** [review]

**Pull request merged**